scripts/game_structure/windows.py

```diff
@@ -458,65 +459,312 @@ def process_event(self, event):
+                                                            scale(pygame.Rect(
+                                                                (550, 410), (200, 60))),
+                                                            object_id="#text_box_30_horizcenter_spacing_95",
+                                                            manager=MANAGER,
+                                                            container=self)
+        self.checkbox_label["singular_label"] = pygame_gui.elements.UITextBox("Singular",
+                                                            scale(pygame.Rect(
+                                                                (255, 570), (200, 60))),
+                                                            object_id="#text_box_30_horizcenter_spacing_95",
+                                                            manager=MANAGER,
+                                                            container=self)
+        self.checkbox_label["plural_label"] = pygame_gui.elements.UITextBox("Plural",
+                                                            scale(pygame.Rect(
+                                                                (470, 570), (200, 60))),
+                                                            object_id="#text_box_30_horizcenter_spacing_95",
+                                                            manager=MANAGER,
+                                                            container=self)
+
+        
+        # Row 1
+        self.boxes["subject"] = pygame_gui.elements.UITextEntryLine(
+            scale(pygame.Rect((180, 290), (200, 60))),
+            placeholder_text=self.the_cat.pronouns[0]['subject'],
+            manager=MANAGER,
+            container=self)
+
+        self.boxes["object"] = pygame_gui.elements.UITextEntryLine(
+            scale(pygame.Rect((430, 290), (200, 60))),
+            placeholder_text=self.the_cat.pronouns[0]['object'],
+            manager=MANAGER,
+            container=self)
+
+        # Row 2
+        self.boxes["poss"] = pygame_gui.elements.UITextEntryLine(
+            scale(pygame.Rect((50, 470), (200, 60))),
+            placeholder_text=self.the_cat.pronouns[0]['poss'],
+            manager=MANAGER,
+            container=self)
+
+        self.boxes["inposs"] = pygame_gui.elements.UITextEntryLine(
+            scale(pygame.Rect((300, 470), (200, 60))),
+            placeholder_text=self.the_cat.pronouns[0]['inposs'],
+            manager=MANAGER,
+            container=self)
+
+        self.boxes["self"] = pygame_gui.elements.UITextEntryLine(
+            scale(pygame.Rect((550, 470), (200, 60))),
+            placeholder_text=self.the_cat.pronouns[0]['self'],
+            manager=MANAGER,
+            container=self)
+        
+        # Save Confirmation
+        self.pronoun_added = pygame_gui.elements.UITextBox(f"Pronoun saved and added to {self.the_cat.name}!",
```

Comment:
When you click the "add new pronoun" button, I would prefer if it only added it as a pronoun option, rather than also adding it to the cat.  This allows people to add a bunch of custom pronouns that they wish to use later without having to go through the remove it from the cat. Additionally, it would mean you no longer have to pass the cat object as a parameter into the window constructor.